roles/python/tasks/main.yml

```diff
@@ -0,0 +1,5 @@
+---
+# Basic sequence of tasks for setting up a Python app.
+
+- include: install_python.yml
+- include: install_dependencies.yml
```

Comment:
What's the value of splitting these out into separate tasks?  Seems potentially confusing to have to look for different files than main tasks; it isn't like the tasks are very long.

Reply:
I mentioned this to Francis on our call – it makes a little less sense here than it might otherwise, but I want to follow this structure for most roles that I write. The purpose doesn't make sense in isolation (the role will always just run the tasks in sequence in `main.yml`) but makes sense when you think about roles as collections of functionality. The idea is that you can import the tasks that you need (almost like methods) in the context of a playbook or another role using [`tasks_from`](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/include_role_module.html#parameter-tasks_from), with the intent that if a playbook needed to run only some parts of a role, it'd be easy to do (but impossible if they're all in `main.yml`). A more real-world example would be running django-compressor tasks: that's a separate task that would fall under the `django` role, but doesn't belong in the `main.yml` sequence, and would be called using `tasks_from` by only the playbooks that need it.